modules/concepts/controllers/admin-controllers/_index.md

```diff
@@ -9,75 +9,179 @@ Using modern pages, you will have access to the PrestaShop debug toolbar, the se
+- `$this->getCurrencyContext()` - Currency context
+- `$this->getCountryContext()` - Country context
+- `$this->dispatchCommand()` - Execute CQRS commands
+- `$this->dispatchQuery()` - Execute CQRS queries
+- `$this->dispatchHookWithParameters()` - Dispatch hooks
+- `$this->presentGrid()` - Present grid data
+
+See the complete list in the [PrestaShopAdminController source code](https://github.com/PrestaShop/PrestaShop/blob/9.0.0/src/PrestaShopBundle/Controller/Admin/PrestaShopAdminController.php).
+{{% /notice %}}
+
+{{% notice warning %}}
+**Doctrine repositories must be injected:**
+
+Doctrine ORM is not directly accessible via `$this->getDoctrine()` or similar methods. All Doctrine repositories and entity managers must be injected as dependencies in your controller's constructor or action methods.
+
+Example:
+```php
+use Doctrine\ORM\EntityManagerInterface;
+use App\Repository\ProductRepository;
+
+public function __construct(
+    private readonly EntityManagerInterface $entityManager,
+    private readonly ProductRepository $productRepository,
+) {
 }
 ```
+{{% /notice %}} 
+
+## Service Configuration
+
+In PrestaShop 9.0, controllers must be defined as services. You have two main approaches to configure your controller service:
+
+### Option 1: Explicit service configuration with tags
 
-If you want Symfony Dependency Injection to inject services into your controller, you need to use specific YAML service declaration:
 ```yaml
+# modules/your-module/config/services.yml
 services:
-  # The name of the service must match the full namespace class
   MyModule\Controller\DemoController:
-    class: MyModule\Controller\DemoController
-    arguments:
-      - '@doctrine.cache.provider'
+    autowire: true
+    autoconfigure: true
+    tags:
+      - { name: controller.service_arguments }
 ```
 
-You can also retrieve services with the container available in symfony controllers ->
-```php
-<?php
-// modules/your-module/src/Controller/DemoController.php
+### Option 2: Using service defaults
 
-namespace MyModule\Controller;
+```yaml
+# modules/your-module/config/services.yml
+services:
+  _defaults:
+    public: false
+    autowire: true
+    autoconfigure: true
+  
+  MyModule\Controller\DemoController: ~
+```
 
-use Doctrine\Common\Cache\CacheProvider;
-use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;
+{{% notice warning %}}
+**Service configuration is mandatory:**
 
-class DemoController extends FrameworkBundleAdminController
-{
-    public function demoAction()
-    {
-        // you can also retrieve services directly from the container
-        $cache = $this->container->get('doctrine.cache');
-        
-        return $this->render('@Modules/your-module/templates/admin/demo.html.twig');
-    }
-}
-```
+One of the two service configuration options above is **essential and required** for your controller to work properly. Without this configuration:
+- **The controller will not function** and will throw errors
```

Comment:
```suggestion
- **The controller will not work** and will throw errors
```
I'm not sure which one is more correct in english, maybe I'm biased with my french point of view 😅